app-libs/stf/src/stf_sgx.rs

```diff
@@ -240,6 +247,43 @@ where
 	}
 }
 
+impl<Call, Getter, State, Runtime> EncointerSchedulerPalletInterface<State>
+	for Stf<Call, Getter, State, Runtime>
+where
+	State: SgxExternalitiesTrait,
+	Runtime: frame_system::Config
+		+ pallet_encointer_scheduler::Config
+		+ pallet_encointer_ceremonies::Config,
+{
+	fn update_ceremony_phase(state: &mut State, next_ceremony_phase: &Vec<u8>) {
+		state.execute_with(|| {
+			let current_ceremony_phase = current_ceremony_phase();
+			match CeremonyPhaseType::decode(&mut next_ceremony_phase.as_slice()) {
+				Ok(decoded_next_ceremony_phase) => {
+					if current_ceremony_phase.is_none()
+						|| decoded_next_ceremony_phase != current_ceremony_phase.unwrap()
```

Comment:
Not sure if it's good to have an unwrap here. Are you sure it's safe?

Reply:
I think so. I check before that it is not None

Reply:
Two remarks then:
1. This is a node storage call via web socket, correct? So the node state may have changed / something else may have gone wrong if you call this again. So I'd argue it's not safe.
2. If you dealt with it before, you're retrieving the same value twice via web socket. That's a performance problem that should be avoidable (not sure how easy, though).

It's fine for this PR, but could you create the issue, if I'm not wrong about it being a ws call?

Reply:
Ahh, forget it. Just saw before meant one line above

Reply:
Not sure, if it's safe, though because I don't think it's guaranteed that `.is_none()` comes before the `unwrap()`.